### PR TITLE
feat: regex pattern additional syntax support

### PR DESCRIPTION
**Pull request by webbdays**
fancy_regex crate provides regex pattern additional syntax support for features like look around.

### CODE REVIEW

**Comment by webbdays** [comment]
please wait.
i will lint check.

File: src/runner.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ impl Requirements {
         let tests: Vec<_> = utils::iterate_directory(&self.root)
             .filter_map(|path_res| {
                 let path = path_res.expect("error while iterating directory");
-                if re.is_match(path.as_str()) {
+                if re.is_match(path.as_str()).unwrap_or_else(|_| panic!("Problem with matching following pattern : '{}'", self.pattern)) {
```
Comment (sunshowers) [inline src/runner.rs:61]: This should also include the path being matched against, and should be phrased as something like:

```
error matching pattern '{}' against path '{}'
```

**Review by sunshowers — COMMENTED** [review]
Thanks!

File: src/runner.rs (hunk 1 above)
Comment (webbdays) [reply src/runner.rs:61]: ok.

File: tests/example.rs [hunk 2]
```diff
@@ -20,8 +20,8 @@ fn test_artifact_utf8(path: &Utf8Path) -> Result<()> {
 datatest_stable::harness!(
     test_artifact,
     "tests/files",
-    r"^.*/*",
+    r"^.*(?<!\.skip)\.txt",
```
Comment (sunshowers) [inline tests/example.rs:23]: Could you make one of them a regular regex? That way we get to see both of the outcomes.

File: src/runner.rs [hunk 3]
```diff
@@ -52,13 +52,13 @@ impl Requirements {
...
     fn expand(&self) -> Vec<Trial> {
-        let re = regex::Regex::new(&self.pattern)
+        let re = fancy_regex::Regex::new(&self.pattern)
             .unwrap_or_else(|_| panic!("invalid regular expression: '{}'", self.pattern));
 
         let tests: Vec<_> = utils::iterate_directory(&self.root)
             .filter_map(|path_res| {
                 let path = path_res.expect("error while iterating directory");
-                if re.is_match(path.as_str()) {
+                if re.is_match(path.as_str()).unwrap_or_else(|_| panic!("error matching pattern '{}' against path '{}'", self.pattern, path.as_str())) {
```
Comment (sunshowers) [inline src/runner.rs:61]:
```suggestion
                if re.is_match(path.as_str()).unwrap_or_else(|error| panic!("error matching pattern '{}' against path '{}': {error}", self.pattern, path.as_str())) {
```

Comment (webbdays) [reply src/runner.rs]: ok

Comment (webbdays) [reply src/runner.rs]: pushed.

**Comment by webbdays** [comment]
Now there is a problem with rust version.

**Comment by webbdays** [comment]
should i use version of fancy_rust for rust 1.65.0 ?

**Comment by sunshowers** [comment]
Thanks, I think it's fine to bump to 1.66. Our promise is for the last 6 months of Rust versions.

**Comment by webbdays** [comment]
ok.

**Comment by webbdays** [comment]
done.

**Comment by webbdays** [comment]
@sunshowers 
updated ci

**Comment by webbdays** [comment]
I think all are fixed by now.

**Comment by sunshowers** [comment]
Thanks! Great work.

**Comment by sunshowers** [comment]
I'll get this out tomorrow.

**Comment by webbdays** [comment]
ok.

**Comment by webbdays** [comment]
Thanks.

**Pull request merged**